# Catalogues exploration
Ce notebook a pour but d'analyser et de rassembler les diffrentes sources de données afin de constituer un catalogue qui se veut être assez exhaustif accessible via une Web API.  

## 1. OpenNGC catalogue
 Ce catalogue a été télécharger depuis [ce lien](https://github.com/mattiaverga/OpenNGC). Il contient toutes les données présentes dans le catalogue NGC.

In [36]:
#import sys
import os
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', 500)
CHEMIN_SOURCES = os.path.join("..","1_Data_brute")

In [2]:
# Lecture de données depuis le fichier source vers un objet DataFrame
col_names = ['NGC_IC_designation', 
    'object_type',
    'Right_Ascension',
    'Declinaison',
    'Constelation_name_abrv',
    'major_axis',
    'minor_axis',
    'major__axis_position_angle', 
    'B_Apparent_Magnitude',
    'V_Apparent_Magnitude',             
    'J_Apparent_Magnitude',
    'H_Apparent_Magnitude',
    'K_Apparent_Magnitude',
    'Mean_surface_brigthness', 
    'Hubble_morphological_type',
    'center_star_U_maghitude',
    'center_star_B_maghitude',
    'center_star_V_maghitude',
    'Messier_number',
    'NGC_numner',
    'IC_number',
    'Center_star_name',
    'Identifiers',
    'Common_names',       
    'NED_notes',  
    'OpenNGC_notes']

NGC_mattiaAverga_GitHub = pd.read_csv(os.path.join(CHEMIN_SOURCES, "NGC_mattiaAverga_GitHub.csv"), sep=';',header=0, names=col_names)
NGC_mattiaAverga_GitHub.shape

(13958, 26)

Dans cette [source](http://messier.obspm.fr/m-q&a.html), on peut lire:
- M24: the Milky Way Patch in Sagittarius; it contains however the 11th magnitude cluster NGC 6603 which is sometimes erroneously listed as M24. Also, it may be that IC 4715 is M24. 
- M25: this is IC 4725, 
- M40: the double star Winnecke 4
- M45: the Pleiades; however, this cluster is associated with nebulae which have NGC numbers.

In [6]:
NGC_mattiaAverga_GitHub.loc[(NGC_mattiaAverga_GitHub['NGC_IC_designation']=='IC4725') |
                        (NGC_mattiaAverga_GitHub['NGC_IC_designation']=='IC4715')]

NGC_IC_designation object_type Right_Ascension  Declinaison  \
4923             IC4715        *Ass     18:16:56.12  -18:30:52.4   
4933             IC4725         OCl     18:31:46.77  -19:06:53.8   

     Constelation_name_abrv  major_axis  minor_axis  \
4923                    Sgr       120.0        60.0   
4933                    Sgr        14.1         NaN   

      major__axis_position_angle  B_Apparent_Magnitude  V_Apparent_Magnitude  \
4923                         NaN                   NaN                   4.5   
4933                         NaN                  5.29                   4.6   

      J_Apparent_Magnitude  H_Apparent_Magnitude  K_Apparent_Magnitude  \
4923                   NaN                   NaN                   NaN   
4933                   NaN                   NaN                   NaN   

      Mean_surface_brigthness Hubble_morphological_type  \
4923                      NaN                       NaN   
4933                      NaN                       NaN   

      center_star_U_maghitude  center_star_B_maghitude  \
4923                      NaN                      NaN   
4933                      NaN                      NaN   

      center_star_V_maghitude  Messier_number NGC_numner IC_number  \
4923                      NaN            24.0        NaN       NaN   
4933                      NaN            25.0        NaN       NaN   

     Center_star_name Identifiers          Common_names  \
4923              NaN         NaN  Small Sgr Star Cloud   
4933              NaN         NaN                   NaN   

                  NED_notes                             OpenNGC_notes  
4923  Milky Way star cloud.  V-mag taken from HEASARC's messier table  
4933                    NaN                                       NaN

*Note:* Les objets M24 et  M25 existent dans NGC_mattiaAverga_GitHub. Mais M40 et M45 n'y sont pas. Ils sont dans l'Addendum.

### Addendum
Le fichier Addendum contient des objets qui ne font partie des catalogue NGC ou IC mais qui peuvent interessé les amateurs de l'astronomie. Les objets M40 et M45 du catalogue de Messier n'ont pas d'identifiant NGC donc ils sont présents dans cet addendum (voir [source](https://github.com/mattiaverga/OpenNGC/blob/master/README.md)).

In [3]:
# Lecture de données depuis le fichier source vers un objet DataFrame
addendum_NGC_mattiaAverga_GitHub = pd.read_csv(os.path.join(CHEMIN_SOURCES, "addendum_NGC_mattiaAverga_GitHub.csv"), sep=';', header=0, names=col_names)
addendum_NGC_mattiaAverga_GitHub.shape

(20, 26)

## 2. Construction du catalogue de Messier
Dans ce qui suit, nous allons construire notre catalogue de Messier à partir de plusieurs sources.

### 2.1. Le catalogue de Messier: *'Datastro'*
[Source](https://www.datastro.eu/explore/dataset/catalogue-de-messier/table/?disjunctive.objet&disjunctive.mag&disjunctive.english_name_nom_en_anglais&disjunctive.french_name_nom_francais&disjunctive.latin_name_nom_latin&sort=messier)

#### 2.1.1. Lecture du fichier

In [4]:
# Lecture de données depuis le fichier source vers un objet DataFrame
col_names = ['Messier_number',
    'NGC_IC_designation', 
    'object_type',
    'Season',
    'Magnitude',
    'Constellation_EN',
    'Constellation_FR',
    'Constellation_Latin',
    'Right_Ascension',        
    'Declinaison',
    'Distance_light_year',
    'Size',
    'Discoverer',
    'Year',
    'Image1',
    'Image2',
    'Constelation_abr' ]
messier_Datastro_website = pd.read_csv(os.path.join(CHEMIN_SOURCES, "messier_Datastro_website.csv"), sep=';', header=0, names=col_names)
messier_Datastro_website.shape

(110, 17)

In [5]:
# Corrections d'un identifiant NGC 
messier_Datastro_website.loc[messier_Datastro_website['NGC_IC_designation']=='NGC 9176', 'NGC_IC_designation'] = 'NGC 1976'

#### 2.1.2. Division du catalogue de Messier

Nous avons besoin de traiter à part les deux objets M40 et M45. Car ces deux objets vont être mergés avec les données présentes dans le *Addendum* et le reste des objets Messier vont être mergés avec les données du *NGC_mattiaAverga_GitHub*.

In [7]:
messier_Datastro_website_without_M40_M45 = messier_Datastro_website[(messier_Datastro_website['Messier_number']!='M40') &
                        (messier_Datastro_website['Messier_number']!='M45')].copy()

messier_Datastro_website_M40_M45 = messier_Datastro_website[(messier_Datastro_website['Messier_number']=='M40') |
                        (messier_Datastro_website['Messier_number']=='M45')].copy()

#### 2.1.3. Merging entre Messier et NGC

In [38]:
# Normalisation du NGC_designation: NGCnnnn (suppression des vides)
messier_Datastro_website_without_M40_M45.loc[messier_Datastro_website_without_M40_M45['NGC_IC_designation'].str.len()==8
                             , 'NGC_IC_designation']= messier_Datastro_website_without_M40_M45['NGC_IC_designation'].str.replace(" ", "")

messier_Datastro_website_without_M40_M45.loc[messier_Datastro_website_without_M40_M45['NGC_IC_designation'].str.len()==7
                             , 'NGC_IC_designation']= messier_Datastro_website_without_M40_M45['NGC_IC_designation'].str.replace(" ", "0")

In [39]:
# Merge du dataframe qui contient tout les objets sauf M40 et M45
merge_messier_Datastro_website_without_M40_M45 = messier_Datastro_website_without_M40_M45.merge(NGC_mattiaAverga_GitHub, how='left', on='NGC_IC_designation')
merge_messier_Datastro_website_without_M40_M45.shape

(108, 42)

*Liste des features:*

- Messier_number_x
- NGC_IC_designation
- *object_type_x* (delete object_type_y is abreviation) 
- Season                         
- Magnitude                       
- Constellation_EN               
- Constellation_FR               
- Constellation_Latin            
- Right_Ascension_x  1 Nan(keep Right_Ascension_y no nan)            
- Declinaison_x  1 Nan(keep Declinaison_y no nan)                
- Distance_light_year           
- Size                           
- Discoverer                     
- Year                          
- Image1                         
- Image2                         
- *Constelation_abr (Delete)*             
- *object_type_y  (delete)*            
- *Right_Ascension_y*             
- *Declinaison_y*                  
- Constelation_name_abrv      
- major_axis                   
- minor_axis                    
- major__axis_position_angle    
- B_Apparent_Magnitude          
- V_Apparent_Magnitude          
- J_Apparent_Magnitude          
- H_Apparent_Magnitude          
- K_Apparent_Magnitude          
- Mean_surface_brigthness       
- Hubble_morphological_type      
- center_star_U_maghitude       
- center_star_B_maghitude       
- center_star_V_maghitude       
- *Messier_number_y (delete)*             
- NGC_numner  (6 double-listed in NGC)                  
- *IC_number (delete: 108 Nan)*                    
- Center_star_name               
- Identifiers                    
- Common_names                   
- NED_notes                      
- OpenNGC_notes

In [40]:
# Suppression des colunne inutiles ou redondantes
drop_list = ['object_type_y', 'Right_Ascension_x', 'Declinaison_x', 'Messier_number_y', 'IC_number']
rename_lis = {'Messier_number_x' : 'Messier_number',
             'object_type_x' : 'object_type',
             'Right_Ascension_y' : 'Right_Ascension',
             'Declinaison_y' : 'Declinaison'}

merge_messier_Datastro_website_without_M40_M45 = merge_messier_Datastro_website_without_M40_M45.drop(columns=drop_list)
merge_messier_Datastro_website_without_M40_M45 = merge_messier_Datastro_website_without_M40_M45.rename(columns=rename_lis)
merge_messier_Datastro_website_without_M40_M45.shape

(108, 37)

In [11]:
# Extraction des objets M40 et M45 depuis l'Addendum
Open_NGC_M40_M45 = addendum_NGC_mattiaAverga_GitHub[~addendum_NGC_mattiaAverga_GitHub['Messier_number'].isna()].copy()
# Ajout du Numero Messier standardisé
Open_NGC_M40_M45['Messier_number_bis'] = 'M'+Open_NGC_M40_M45['Messier_number'].astype('int').astype('str')

# Remplacement de la colonne Messier_number par la nouvelle colonne
Open_NGC_M40_M45 = Open_NGC_M40_M45.drop(columns='Messier_number')
Open_NGC_M40_M45 = Open_NGC_M40_M45.rename(columns={'Messier_number_bis': 'Messier_number'})
Open_NGC_M40_M45

NGC_IC_designation object_type Right_Ascension Declinaison  \
13               M040          **      12:22:16.1   +58:05:04   
14             Mel022         OCl      03:47:28.6   +24:06:19   

   Constelation_name_abrv  major_axis  minor_axis  major__axis_position_angle  \
13                    UMa         NaN         NaN                         NaN   
14                    Tau       150.0       150.0                        90.0   

    B_Apparent_Magnitude  V_Apparent_Magnitude  J_Apparent_Magnitude  \
13                   NaN                   8.0                   NaN   
14                   NaN                   1.2                   NaN   

    H_Apparent_Magnitude  K_Apparent_Magnitude  Mean_surface_brigthness  \
13                   NaN                   NaN                      NaN   
14                   NaN                   NaN                      NaN   

   Hubble_morphological_type  center_star_U_maghitude  \
13                       NaN                      NaN   
14                       NaN                      NaN   

    center_star_B_maghitude  center_star_V_maghitude  NGC_numner  IC_number  \
13                      NaN                      NaN         NaN        NaN   
14                      NaN                      NaN         NaN        NaN   

    Center_star_name        Identifiers Common_names NED_notes  \
13               NaN  WDS J12222+5805AB          NaN       NaN   
14               NaN          MWSC 0305     Pleiades       NaN   

                                        OpenNGC_notes Messier_number  
13            VMag taken from HEASARC's messier table            M40  
14  Coordinates taken from Simbad, VMag taken from...            M45

In [12]:
# Merge du dataframe qui contient M40 et M45
merge_messier_Datastro_website_M40_M45 = messier_Datastro_website_M40_M45.merge(Open_NGC_M40_M45, how='left', on='Messier_number')
merge_messier_Datastro_website_M40_M45

Messier_number NGC_IC_designation_x                object_type_x  \
0            M45                  NaN   Open Cluster / Amas Ouvert   
1            M40           Winnecke 4  Double star / Étoile Double   

               Season  Magnitude Constellation_EN Constellation_FR  \
0      Winter / Hiver          1              NaN              NaN   
1  Spring / Printemps          9              NaN              NaN   

  Constellation_Latin Right_Ascension_x Declinaison_x  Distance_light_year  \
0                 NaN               NaN           NaN                410.0   
1                 NaN               NaN           NaN                  NaN   

     Size Discoverer    Year                                Image1  \
0  120,0'        NaN     NaN  http://www.lasam.ca/messier/M045.JPG   
1     NaN   Hevelius  1660.0  http://www.lasam.ca/messier/M040.JPG   

                                              Image2 Constelation_abr  \
0  https://www.datastro.eu/api/v2/catalog/dataset...              NaN   
1  https://www.datastro.eu/api/v2/catalog/dataset...              NaN   

  NGC_IC_designation_y object_type_y Right_Ascension_y Declinaison_y  \
0               Mel022           OCl        03:47:28.6     +24:06:19   
1                 M040            **        12:22:16.1     +58:05:04   

  Constelation_name_abrv  major_axis  minor_axis  major__axis_position_angle  \
0                    Tau       150.0       150.0                        90.0   
1                    UMa         NaN         NaN                         NaN   

   B_Apparent_Magnitude  V_Apparent_Magnitude  J_Apparent_Magnitude  \
0                   NaN                   1.2                   NaN   
1                   NaN                   8.0                   NaN   

   H_Apparent_Magnitude  K_Apparent_Magnitude  Mean_surface_brigthness  \
0                   NaN                   NaN                      NaN   
1                   NaN                   NaN                      NaN   

  Hubble_morphological_type  center_star_U_maghitude  center_star_B_maghitude  \
0                       NaN                      NaN                      NaN   
1                       NaN                      NaN                      NaN   

   center_star_V_maghitude  NGC_numner  IC_number  Center_star_name  \
0                      NaN         NaN        NaN               NaN   
1                      NaN         NaN        NaN               NaN   

         Identifiers Common_names NED_notes  \
0          MWSC 0305     Pleiades       NaN   
1  WDS J12222+5805AB          NaN       NaN   

                                       OpenNGC_notes  
0  Coordinates taken from Simbad, VMag taken from...  
1            VMag taken from HEASARC's messier table

In [13]:
# Move data before deleting NGC_IC_designation_y
merge_messier_Datastro_website_M40_M45.loc[merge_messier_Datastro_website_M40_M45['NGC_IC_designation_x'].isna(),
                                          'NGC_IC_designation_x'] = merge_messier_Datastro_website_M40_M45['NGC_IC_designation_y']

In [14]:
# Suppression des colunne inutiles ou redondantes
drop_list = ['NGC_IC_designation_y', 'object_type_y', 'Right_Ascension_x', 'Declinaison_x', 'IC_number']
rename_lis = {'NGC_IC_designation_x': 'NGC_IC_designation',
             'object_type_x' : 'object_type',
             'Right_Ascension_y' : 'Right_Ascension',
             'Declinaison_y' : 'Declinaison'}

merge_messier_Datastro_website_M40_M45 = merge_messier_Datastro_website_M40_M45.drop(columns=drop_list)
merge_messier_Datastro_website_M40_M45 = merge_messier_Datastro_website_M40_M45.rename(columns=rename_lis)
merge_messier_Datastro_website_M40_M45.shape

(2, 37)

#### 2.1.4. Concatination entre Messier contenant les objets M40 et M45 et le Messier ayant le reste des objets

In [15]:
messier_catalogue = pd.concat([merge_messier_Datastro_website_without_M40_M45, merge_messier_Datastro_website_M40_M45])

In [16]:
messier_catalogue[messier_catalogue['Hubble_morphological_type'].isna()]

Messier_number NGC_IC_designation  \
2              M76            NGC0650   
3              M80            NGC6093   
4              M93            NGC2447   
7              M37            NGC2099   
8              M36            NGC1960   
9              M48            NGC2548   
14             M16            NGC6611   
15              M4            NGC6121   
16             M15            NGC7078   
17             M18            NGC6613   
18             M21            NGC6531   
19             M35            NGC2168   
21             M73            NGC6994   
23             M78            NGC2068   
26             M69            NGC6637   
27             M57            NGC6720   
29             M44            NGC2632   
31             M72            NGC6981   
32              M3            NGC5272   
33              M7            NGC6475   
34              M5            NGC5904   
35             M39            NGC7092   
36             M29            NGC6913   
37            M107            NGC6171   
39             M34            NGC1039   
40             M56            NGC6779   
43              M9            NGC6333   
44              M1            NGC1952   
45             M22            NGC6656   
46             M30            NGC7099   
..             ...                ...   
60             M68            NGC4590   
63             M47            NGC2422   
65             M79            NGC1904   
67             M20            NGC6514   
68             M10            NGC6254   
69             M14            NGC6402   
70             M24            NGC6603   
73             M75            NGC6864   
75             M41            NGC2287   
76             M46            NGC2437   
80              M8            NGC6523   
81             M11            NGC6705   
82             M23            NGC6494   
83             M25             IC4725   
84             M54            NGC6715   
85             M62            NGC6266   
87             M67            NGC2682   
88             M71            NGC6838   
91            M103            NGC0581   
93             M70            NGC6681   
94             M50            NGC2323   
97             M97            NGC3587   
99             M92            NGC6341   
100            M90            NGC4560   
102             M2            NGC7089   
104            M17            NGC6618   
105            M27            NGC6853   
106            M28            NGC6626   
0              M45             Mel022   
1              M40         Winnecke 4   

                                   object_type              Season  Magnitude  \
2      Planetary Nebula / Nébuleuse Planétaire    Autumn / Automne         10   
3           Globular Cluster / Amas Globulaire        Summer / Été          7   
4                   Open Cluster / Amas Ouvert      Winter / Hiver          6   
7                   Open Cluster / Amas Ouvert      Winter / Hiver          5   
8                   Open Cluster / Amas Ouvert      Winter / Hiver          6   
9                   Open Cluster / Amas Ouvert      Winter / Hiver          5   
14      Emission Nebula / Nébuleuse à émission        Summer / Été          6   
15          Globular Cluster / Amas Globulaire        Summer / Été          5   
16          Globular Cluster / Amas Globulaire    Autumn / Automne          6   
17                  Open Cluster / Amas Ouvert        Summer / Été          6   
18                  Open Cluster / Amas Ouvert        Summer / Été          5   
19                  Open Cluster / Amas Ouvert      Winter / Hiver          5   
21                  Open Cluster / Amas Ouvert    Autumn / Automne          9   
23   Reflection Nebula / Nébuleuse à réflexion      Winter / Hiver          8   
26          Globular Cluster / Amas Globulaire        Summer / Été          7   
27     Planetary Nebula / Nébuleuse Planétaire        Summer / Été          9   
29                  Open Cluster / Amas Ouvert  Spring / Printemps          3   
31 

### 2.3. Le catalogue de Messier: *'jbcurtin_gitHub'*
Ce catalogue ([Voir référence](https://github.com/jbcurtin/messier-catalogue)) est pricipalement intéressant pour les images qu'il contient pour chaque objet.

In [17]:
# Lecture de données depuis le fichier source vers un objet DataFrame
col_names = ['Messier_number_jbcurtin_gitHub',
    'NGC_IC_designation_jbcurtin_gitHub', 
    'Common_name_jbcurtin_gitHub',
    'image_jbcurtin_gitHub',
    'Object_type_jbcurtin_gitHub',
    'Distance_KLY_jbcurtin_gitHub',             
    'Constellation_jbcurtin_gitHub',
    'Apparent_magnitude_jbcurtin_gitHub',
    'Right_Ascension_jbcurtin_gitHub',        
    'Declinaison_jbcurtin_gitHub']
messier_with_picture_links_jbcurtin_gitHub = pd.read_csv(os.path.join(CHEMIN_SOURCES, "messier_with_picture_links_jbcurtin_gitHub.csv"), keep_default_na=True, sep=';', header=0, names=col_names)
messier_with_picture_links_jbcurtin_gitHub.loc[messier_with_picture_links_jbcurtin_gitHub['Common_name_jbcurtin_gitHub']==' ',
                       'Common_name_jbcurtin_gitHub'] = np.nan #not a space
    
messier_with_picture_links_jbcurtin_gitHub.dtypes

Messier_number_jbcurtin_gitHub         object
NGC_IC_designation_jbcurtin_gitHub     object
Common_name_jbcurtin_gitHub            object
image_jbcurtin_gitHub                  object
Object_type_jbcurtin_gitHub            object
Distance_KLY_jbcurtin_gitHub           object
Constellation_jbcurtin_gitHub          object
Apparent_magnitude_jbcurtin_gitHub    float64
Right_Ascension_jbcurtin_gitHub        object
Declinaison_jbcurtin_gitHub            object
dtype: object

In [18]:
# Merge entre 'messier_catalogue' et le nouveau dataset
messier_catalogue2 = messier_with_picture_links_jbcurtin_gitHub.merge(messier_catalogue,how='left', left_on='Messier_number_jbcurtin_gitHub', right_on='Messier_number')
messier_catalogue2.shape

(110, 47)

In [19]:
#keep 'Common_name_jbcurtin_gitHub', delete_Common_names
messier_catalogue2.loc[messier_catalogue2['Common_name_jbcurtin_gitHub'].isna(),
                   'Common_name_jbcurtin_gitHub'] = messier_catalogue2['Common_names']

# In 'Object_type_jbcurtin_gitHub', Galaxy is subtyped. So we should keep that info
# Keep 'object_type', delete 'Object_type_jbcurtin_gitHub'
messier_catalogue2.loc[messier_catalogue2['object_type'].str.contains('Galaxy'),
                   'object_type'] = messier_catalogue2['Object_type_jbcurtin_gitHub']

#Delete 'Constellation_jbcurtin_gitHub'
messier_catalogue2.loc[messier_catalogue2['Constellation_Latin'].isna(),
                   'Constellation_Latin'] = messier_catalogue2['Constellation_jbcurtin_gitHub']

drop_list = ['Common_names', 'Messier_number_jbcurtin_gitHub', 'NGC_IC_designation_jbcurtin_gitHub', 
             'Object_type_jbcurtin_gitHub', 'Distance_KLY_jbcurtin_gitHub', 'Constellation_jbcurtin_gitHub',
            'Right_Ascension_jbcurtin_gitHub', 'Declinaison_jbcurtin_gitHub']
rename_lis = {'Common_name_jbcurtin_gitHub': 'Common_name'}

messier_catalogue2 = messier_catalogue2.drop(columns=drop_list)
messier_catalogue2 = messier_catalogue2.rename(columns=rename_lis)
messier_catalogue2.shape


(110, 39)

### 2.4. Le catalogue de Messier: *'Nexstarsite'*

https://www.nexstarsite.com/Book/DSO.htm

In [20]:
MessierObjects_nexstarsite = pd.read_excel(os.path.join(CHEMIN_SOURCES, "nexstarsite", "MessierObjects_nexstarsite.xls"), skiprows=2)
MessierObjects_nexstarsite.dtypes

ObjectNum          int64
Name              object
Type              object
Constellation     object
RAHour             int64
RAMinute         float64
DecSign           object
DecDeg             int64
DecMinute        float64
Magnitude         object
Info              object
Distance (ly)      int64
dtype: object

Ce dataset ne contient pas de nouvelles données.

In [21]:
messier_catalogue_final = messier_catalogue2

In [22]:
messier_catalogue_final.dtypes

Common_name                            object
image_jbcurtin_gitHub                  object
Apparent_magnitude_jbcurtin_gitHub    float64
Messier_number                         object
NGC_IC_designation                     object
object_type                            object
Season                                 object
Magnitude                               int64
Constellation_EN                       object
Constellation_FR                       object
Constellation_Latin                    object
Distance_light_year                   float64
Size                                   object
Discoverer                             object
Year                                  float64
Image1                                 object
Image2                                 object
Constelation_abr                       object
Right_Ascension                        object
Declinaison                            object
Constelation_name_abrv                 object
major_axis                        

## 3. Construction du catalogue de Caldwell

In [73]:
# Lecture de données depuis le fichier source vers un objet DataFrame
col_names = ['Caldwell_number',
    'NGC_IC_designation', 
    'object_type',
    'Constelation_abr',
    'Magnitude',
    'Size',
    'Right_Ascension_Hour',
    'Right_Ascension_Minute', 
    'Declinaison_Sign',
    'Declinaison_Deg',
    'Declinaison_Minute'
     ]
#messier_with_picture_links_jbcurtin_gitHub = pd.read_csv(os.path.join(CHEMIN_SOURCES, "messier_with_picture_links_jbcurtin_gitHub.csv"), keep_default_na=True, sep=';', header=0, names=col_names)

CaldwellObjects_nexstarsite = pd.read_excel(os.path.join(CHEMIN_SOURCES, "nexstarsite", "CaldwellObjects_nexstarsite.xls"), skiprows=2, keep_default_na=True, names=col_names)
CaldwellObjects_nexstarsite.shape

(109, 11)

In [74]:
CaldwellObjects_nexstarsite

Caldwell_number NGC_IC_designation        object_type Constelation_abr  \
0                  1            NGC 188       Open Cluster              Cep   
1                  2             NGC 40   Planetary Nebula              Cep   
2                  3           NGC 4236      Spiral Galaxy              Dra   
3                  4           NGC 7023      Bright Nebula              Cep   
4                  5             IC 342      Spiral Galaxy              Cam   
5                  6           NGC 6543   Planetary Nebula              Dra   
6                  7           NGC 2403      Spiral Galaxy              Cam   
7                  8            NGC 559       Open Cluster              Cas   
8                  9            Sh2-155      Bright Nebula              Cep   
9                 10            NGC 663       Open Cluster              Cas   
10                11           NGC 7635      Bright Nebula              Cas   
11                12           NGC 6946      Spiral Galaxy              Cep   
12                13            NGC 457       Open Cluster              Cas   
13                14        NGC 869/884       Open Cluster              Per   
14                15           NGC 6826   Planetary Nebula              Cyg   
15                16           NGC 7243       Open Cluster              Lac   
16                17            NGC 147  Elliptical Galaxy              Cas   
17                18            NGC 185  Elliptical Galaxy              Cas   
18                19            IC 5146      Bright Nebula              Cyg   
19                20           NGC 7000      Bright Nebula              Cyg   
20                21           NGC 4449   Irregular Galaxy              CVn   
21                22           NGC 7662   Planetary Nebula              And   
22                23            NGC 891      Spiral Galaxy              And   
23                24           NGC 1275   Irregular Galaxy              Per   
24                25           NGC 2419   Globular Cluster              Lyn   
25                26           NGC 4244      Spiral Galaxy              CVn   
26                27           NGC 6888      Bright Nebula              Cyg   
27                28            NGC 752       Open Cluster              And   
28                29           NGC 5005      Spiral Galaxy              CVn   
29                30           NGC 7331      Spiral Galaxy              Peg   
..               ...                ...                ...              ...   
79                80           NGC 5139   Globular Cluster              Cen   
80                81           NGC 6352   Globular Cluster              Ara   
81                82           NGC 6193       Open Cluster              Ara   
82                83           NGC 4945      Spiral Galaxy              Cen   
83                84           NGC 5286   Globular Cluster              Cen   
84                85            IC 2391       Open Cluster              Vel   
85                86           NGC 6397   Globular Cluster              Ara   
86                87           NGC 1261   Globular Cluster              Hor   
87                88           NGC 5823       Open Cluster              Cir   
88                89           NGC 6087       Open Cluster              Nor   
89                90           NGC 2867   Planetary Nebula              Car   
90                91           NGC 3532       Open Cluster              Car   
91                92           NGC 3372      Bright Nebula              Car   
92                93           NGC 6752   Globular Cluster              Pav   
93                94           NGC 4755       Open Cluster              Cru   
94                95           NGC 6025       Open Cluster              TrA   
95                96           NGC 2516       Open Cluster              Car   
96                97           NGC 3766       Open Cluster              Cen   
97                98           NGC 4609       Open Clus

In [61]:
NGC_mattiaAverga_GitHub[NGC_mattiaAverga_GitHub['NGC_IC_designation'] =="NGC0104"]

NGC_IC_designation object_type Right_Ascension  Declinaison  \
5724            NGC0104         GCl     00:24:05.36  -72:04:53.2   

     Constelation_name_abrv  major_axis  minor_axis  \
5724                    Tuc        31.8         NaN   

      major__axis_position_angle  B_Apparent_Magnitude  V_Apparent_Magnitude  \
5724                         NaN                  5.78                  4.09   

      J_Apparent_Magnitude  H_Apparent_Magnitude  K_Apparent_Magnitude  \
5724                  2.29                  1.76                  1.54   

      Mean_surface_brigthness Hubble_morphological_type  \
5724                      NaN                       NaN   

      center_star_U_maghitude  center_star_B_maghitude  \
5724                      NaN                      NaN   

      center_star_V_maghitude  Messier_number NGC_numner IC_number  \
5724                      NaN             NaN        NaN       NaN   

     Center_star_name                              Identifiers  \
5724              NaN  2MASX J00240535-7204531,C 106,MWSC 0038   

        Common_names NED_notes OpenNGC_notes  
5724  47 Tuc Cluster       NaN           NaN

In [58]:
addendum_NGC_mattiaAverga_GitHub[addendum_NGC_mattiaAverga_GitHub['NGC_IC_designation'] =="NGC2070"]

Empty DataFrame
Columns: [NGC_IC_designation, object_type, Right_Ascension, Declinaison, Constelation_name_abrv, major_axis, minor_axis, major__axis_position_angle, B_Apparent_Magnitude, V_Apparent_Magnitude, J_Apparent_Magnitude, H_Apparent_Magnitude, K_Apparent_Magnitude, Mean_surface_brigthness, Hubble_morphological_type, center_star_U_maghitude, center_star_B_maghitude, center_star_V_maghitude, Messier_number, NGC_numner, IC_number, Center_star_name, Identifiers, Common_names, NED_notes, OpenNGC_notes]
Index: []

In [48]:
NGC_mattiaAverga_GitHub

NGC_IC_designation object_type Right_Ascension  Declinaison  \
0                 IC0001          **     00:08:27.05  +27:43:03.6   
1                 IC0002           G     00:11:00.88  -12:49:22.3   
2                 IC0003           G     00:12:06.09  -00:24:54.8   
3                 IC0004           G     00:13:26.94  +17:29:11.2   
4                 IC0005           G     00:17:34.93  -09:32:36.1   
5                 IC0006           G     00:18:55.04  -03:16:33.9   
6                 IC0007           G     00:18:53.16  +10:35:40.9   
7                 IC0008           G     00:19:02.72  -03:13:19.5   
8                 IC0009           G     00:19:43.98  -14:07:18.8   
9                 IC0010           G     00:20:17.34  +59:18:13.6   
10                IC0011         Dup     00:52:59.35  +56:37:18.8   
11                IC0012           G     00:20:15.05  -02:39:11.2   
12                IC0013           G     00:20:20.09  +07:42:01.8   
13                IC0014          **     00:22:31.29  +10:29:24.9   
14                IC0015           G     00:27:57.59  -00:03:40.6   
15                IC0016           G     00:28:07.65  -13:05:38.1   
16                IC0017           G     00:28:29.78  +02:38:55.0   
17                IC0018           G     00:28:34.96  -11:35:12.1   
18                IC0019           G     00:28:39.48  -11:38:26.8   
19                IC0020           G     00:28:39.68  -13:00:37.1   
20                IC0021           G     00:29:10.43  -00:09:49.6   
21                IC0022           G     00:29:33.17  -09:04:50.7   
22                IC0023           G     00:30:50.80  -12:43:12.9   
23                IC0024          **     00:31:16.63  +30:50:21.5   
24                IC0025           G     00:31:12.09  -00:24:26.5   
25                IC0026         Dup     00:31:45.94  -13:20:14.9   
26                IC0027           G     00:33:06.24  -13:22:17.4   
27                IC0028           G     00:33:08.72  -13:27:22.6   
28                IC0029           G     00:34:10.80  -02:10:39.5   
29                IC0030           G     00:34:14.72  -02:05:04.5   
...                  ...         ...             ...          ...   
13928            NGC7811           G     00:02:26.47  +03:21:06.9   
13929            NGC7812           G     00:02:54.47  -34:14:08.3   
13930            NGC7813           G     00:04:09.12  -11:59:02.0   
13931            NGC7814           G     00:03:14.89  +16:08:43.5   
13932            NGC7815           *     00:03:24.83  +20:42:15.0   
13933            NGC7816           G     00:03:48.86  +07:28:43.1   
13934            NGC7817           G     00:03:58.91  +20:45:08.4   
13935            NGC7818           G     00:04:08.85  +07:22:45.8   
13936            NGC7819           G     00:04:24.54  +31:28:19.4   
13937            NGC7820           G     00:04:30.79  +05:12:01.0   
13938            NGC7821           G     00:05:16.72  -16:28:36.9   
13939            NGC7822         HII     00:03:35.35  +67:09:41.9   
13940            NGC7823           G     00:04:45.59  -62:03:41.6   
13941            NGC7824           G     00:05:06.23  +06:55:12.4   
13942            NGC7825           G     00:05:06.60  +05:12:13.2   
13943            NGC7826       Other     00:05:19.32  -20:41:17.8   
13944            NGC7827           G     00:05:27.66  +05:13:20.4   
13945            NGC7828           G     00:06:27.07  -13:24:58.0   
13946            NGC7829           G     00:06:28.98  -13:25:14.2   
13947            NGC7830           *     00:06:01.86  +08:20:34.0   
13948            NGC7831           G     00:07:19.53  +32:36:33.3   
13949            NGC7832           G     00:06:28.46  -03:42:58.1   
13950            NGC7833       Other     00:06:31.46  +27:38:22.8   
13951            NGC7834           G     00:06:37.82  +08:22:04.4   
13952            NGC7835           G     00:06:46.77  +08:25:33.4   
13953            NGC7836           G     00:08:01.64  +33:04:14.8   
1395

In [ ]:
# Normalisation du NGC_designation: NGCnnnn (suppression des vides)
CaldwellObjects_nexstarsite.loc[CaldwellObjects_nexstarsite['NGC_IC_designation'].str.len()==8
                             , 'NGC_IC_designation']= CaldwellObjects_nexstarsite['NGC_IC_designation'].str.replace(" ", "")

CaldwellObjects_nexstarsite.loc[CaldwellObjects_nexstarsite['NGC_IC_designation'].str.len()==7
                             , 'NGC_IC_designation']= CaldwellObjects_nexstarsite['NGC_IC_designation'].str.replace(" ", "0")

## 4. Construction du catalogue de Herschel400

In [24]:
Herschel400_nexstarsite = pd.read_excel(os.path.join(CHEMIN_SOURCES, "nexstarsite","Herschel400_nexstarsite.xls"), skiprows=2)
Herschel400_nexstarsite.shape

(400, 11)

# Catalogues Merging

## Final Catalogue file

In [29]:
reorder = ['Messier_number', 'NGC_IC_designation', 'object_type', 'Common_name', 'Constellation_FR', 'Constellation_EN',
'Constellation_Latin', 'Constelation_abr','Constelation_name_abrv','Hubble_morphological_type', 'Discoverer', 'Year', 
'Season', 'Right_Ascension','Declinaison', 'Distance_light_year', 'Size', 'major_axis', 'minor_axis', 'major__axis_position_angle',
'Apparent_magnitude_jbcurtin_gitHub', 'Magnitude', 'B_Apparent_Magnitude', 'V_Apparent_Magnitude', 'J_Apparent_Magnitude',
'H_Apparent_Magnitude', 'K_Apparent_Magnitude', 'Mean_surface_brigthness',
'Center_star_name', 'center_star_U_maghitude', 'center_star_B_maghitude', 'center_star_V_maghitude',
'Image1', 'Image2', 'image_jbcurtin_gitHub', 'NGC_numner', 'Identifiers', 'NED_notes', 'OpenNGC_notes']

messier_catalogue_final = messier_catalogue_final[reorder]

In [35]:
#messier_catalogue_final.to_csv(os.path.join(CHEMIN_OUTPUT, "Final_catalogue.csv"),";")
messier_catalogue_final.to_csv("Final_catalogue.csv",";")